roles/jboss_ws/tasks/main.yml

```diff
@@ -0,0 +1,93 @@
+---
+
+- name: internal_host_started_processing_role
+  set_fact:
+    internal_host_started_processing_role: "jboss_web_server"
+
+# ----------------------------- FIND JWS_HOME ---------------------------------
+
+# If JWS is not on target server, this will only return whatever directory is above TOMCAT_HOME
+- name: search for JWS_HOME in tomcat process
+  raw: ps -ef | grep tomcat | grep java |awk -F"catalina.base=" '{split($2,a," ");print FS a[1]}' |  awk 'BEGIN {FS="/";OFS="/";}{ $1=""; $NF="";print}'
+  register: jws_find_home_from_tomcat_process
+  ignore_errors: yes
+
+  # If client has not renamed default JWS_HOME folder and followed recommended zip file install instructions, JWS_HOME should be here.
+- name: search JWS_HOME in opt folder
+  raw: find /opt -type d | egrep '*(ews|jws)*[0-99]\.[0-99]$'
+  register: jws_find_home_from_opt
+  ignore_errors: yes
+
+- name: combine JWS_HOME candidates into a list
+  set_fact:
+    jws_home_candidates: "{{
+        (jws_find_home_from_opt.get('stdout_lines', []) +
+         jws_find_home_from_tomcat_process.get('stdout_lines', []))|difference(['']) | unique }}"
+  ignore_errors: yes
+
+- name: ls JWS_HOME
+  raw: ls -d1 "{{ item }}" 2>/dev/null
+  register: find_jws_home
+  with_items: "{{ jws_home_candidates }}"
+  ignore_errors: yes
+
+- name: set jws_home based on find_jws_home
+  set_fact:
+      jws_home: "{{ find_jws_home['results'][0]['stdout_lines'][0] }}" 
```

Comment:
I'd suggest adding some guards to this in case you are trying to index into the dictionary or list and its not there.
i.e.
```
when:
  - 'results' in find_jws_home
  - find_jws_home['results'] | length > 0
  - 'stdout_lines' in find_jws_home['results'][0]
  - find_jws_home['results'][0]['stdout_lines'] | length > 1